code/modules/food_and_drink/snacks.dm

```diff
@@ -1196,7 +1196,7 @@
 	initial_reagents = list("sugar" = 20)
 	food_effects = list("food_energized")
 	var/can_add_frosting = TRUE
-	var/list/styles = list("icing", "sprinkles")
+	var/list/frostingstyle = list("icing", "sprinkles", "zigzags", "center fill", "half and half icing", "dipped icing", "heart", "star")
```

Comment:
This makes it so the list exists only once and not on each donut. It also puts the icon states directily into this list, see the next suggestion for how this helps:
```suggestion
	var/static/list/frostingstyle = list(
	"icing" = "icing",
	"sprinkles" = "sprinkles",
	"zigzags" = "zigzags",
	"center fill" = "center",
	"half and half icing" = "half",
	"dipped icing"= "dipped",
	"heart" = "heart",
	"star" = "star)
```

Reply:
Can I also recommend renaming this variable to `frostingstyles` or even `frosting_styles`? Just for readability's sake and also because imo it makes sense for it to be in plural when it contains multiple things.